packages/xy-chart/src/series/PointSeries.jsx

```diff
@@ -73,28 +133,33 @@ export default class PointSeries extends React.PureComponent {
           const defined = isDefined(xVal) && isDefined(yVal);
           const cx = xScale(xVal);
           const cy = yScale(yVal);
-          const circleFill = d.fill || callOrValue(fill, d, i);
+          const computedFill = d.fill || callOrValue(fill, d, i);
           const key = `${label}-${x(d)}-${i}`;
           if (defined && d.label) {
             labels.push({ x: cx, y: cy, label: d.label, key: `${key}-label` });
           }
-          return defined && (
-            <GlyphDot
-              key={key}
-              cx={cx}
-              cy={cy}
-              r={d.size || callOrValue(size, d, i)}
-              fill={circleFill}
-              fillOpacity={d.fillOpacity || callOrValue(fillOpacity, d, i)}
-              stroke={d.stroke || callOrValue(stroke, d, i)}
-              strokeWidth={d.strokeWidth || callOrValue(strokeWidth, d, i)}
-              strokeDasharray={d.strokeDasharray || callOrValue(strokeDasharray, d, i)}
-              onMouseMove={onMouseMove && ((event) => {
-                onMouseMove({ event, data, datum: d, color: circleFill });
-              })}
-              onMouseLeave={onMouseLeave}
-            />
-          );
+          const computedSize = d.size || callOrValue(size, d, i);
+          const computedFillOpacity = d.fillOpacity || callOrValue(fillOpacity, d, i);
+          const computedStroke = d.stroke || callOrValue(stroke, d, i);
+          const computedStrokeWidth = d.strokeWidth || callOrValue(strokeWidth, d, i);
+          const computedStrokeDasharray = d.strokeDasharray || callOrValue(strokeDasharray, d, i);
+          const props = {
+              key: key,
+              x: cx,
+              y: cy,
+              size: computedSize,
+              fill: computedFill,
+              fillOpacity: computedFillOpacity,
+              stroke: computedStroke,
+              strokeWidth: computedStrokeWidth,
+              strokeDasharray: computedStrokeDasharray,
+              onMouseMove,
+              onMouseLeave,
+              data,
+              datum: d,
+          };
+          return defined &&
+            React.createElement(pointComponent, props);
         })}
         {/* Put labels on top */}
         {labels.map(d => React.cloneElement(labelComponent, d, d.label))}
```

Comment:
just noting this is a little strange to mix cloning + creating elements. I think you have it right for the glyph and we should update label sometime soon.